src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -119,6 +119,7 @@ private static bool SupportsCodeGeneration(RegexMethod rm)
                 // Place an artificial limit on max tree depth in order to mitigate such issues.
                 // The allowed depth can be tweaked as needed;its exceedingly rare to find
                 // expressions with such deep trees.
+                reason = "the regex is too complex";
```

Comment:
I know I suggested this text, but thinking about it a bit more, I think we should be more specific:
```suggestion
                reason = "the regex will result in code that may exceed C# compiler limits";
```

Reply:
Done.